src/metax_api/models/catalog_record.py

```diff
@@ -1216,7 +1225,8 @@ def _post_create_operations(self):
                 dvs.save()
                 dvs.records.add(self)
 
-            if get_identifier_type(self.preferred_identifier) == IdentifierType.DOI:
+            if (get_identifier_type(self.preferred_identifier) == IdentifierType.DOI or
+            self.use_doi_for_published is True):
```

Comment:
double indentation for the second line is preferred

Reply:
Ok! Now resolved and also flake is happy with 

            if (get_identifier_type(self.preferred_identifier) == IdentifierType.DOI or
                    self.use_doi_for_published is True):
                self._.....